django/contrib/gis/geos/linestring.py

```diff
@@ -32,18 +32,22 @@ def __init__(self, *args, **kwargs):
         else:
             coords = args
 
+        if not isinstance(coords, (tuple, list, numpy.ndarray)):
+            raise TypeError('Invalid initialization input for LineStrings.')
+
+        ncoords = len(coords)
+        if ncoords < self._minlength:
+            raise TypeError(
+                '%s requires at least %d points, got %s.' % (
+                    self.__class__.__name__,
+                    self._minlength,
+                    ncoords,
+                )
+            )
+
         if isinstance(coords, (tuple, list)):
```

Comment:
It's required, because later behavior is different for list/tuple and for array.